### PR TITLE
IS-17855: Added new version of bidirectional sync and fixed some other links

### PR DESCRIPTION
**Pull request by erik-leven**
image
image
image

### CODE REVIEW

File: hub/bidirectional-synchronization.rst [hunk 1]
```diff
@@ -52,17 +52,113 @@ Internal duplicate management
 
 The concept of internal duplicates covers two different scenarios. Firstly, a system should not be able to insert its own entities back to itself (*management of new duplicates*). Secondly, preexisting duplicates (the sales team might have created two versions of a company in their CRM system) should be treated as the same entity whenever possible (*management of preexisting duplicates*).  
 
-|start-h4| **Management of new duplicates** |end-h4|
+|start-h5| **Management of new duplicates** |end-h5|
 
-foobar
+We can ensure that entities are not inserted into their own source by applying a combination of the :ref:`namespace split pattern <namespace_split>` and the :ref:`duplicate hops block pattern <duplicate-hops-block>`. 
 
+The first will ensure that all entities attempting to communicate with the source system are doing to in the correct semantic context, i.e. they are using the correct namespace in their ``_id`` value. This allows you to block inserts if entities already have the target system's namespace.
```
Comment (djalil-bouhachi) [inline hub/bidirectional-synchronization.rst:59]: Typo " are doing it"

File: hub/bidirectional-synchronization.rst [hunk 2]
```diff
@@ -52,17 +52,113 @@ Internal duplicate management
...
 
+The second will ensure that no entities that already have a successful insert in the sink dataset of the ``-share`` pipe will be inserted again.
 
+|start-h5| **Management of preexisting duplicates** |end-h5|
 
-.. |start-h4| raw:: html
+The only way to ensure that preexisting duplicates do not propagate as duplicates downstream is to identify them and :ref:`merge <merging>` then in their global pipe. This can be done by locating an  appropriate merge criterion that the two entities have in common.
```
Comment (djalil-bouhachi) [inline hub/bidirectional-synchronization.rst:65]: typo: Merge "them" in their global pipe.

File: hub/bidirectional-synchronization.rst [hunk 3]
```diff
@@ -52,17 +52,113 @@ Internal duplicate management
...
+External duplicate management
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-.. |end-h4| raw:: html
+External duplicate management covers two different scenarios:
+
+1. Ensuring that preexisting duplicates are not propagated as duplicates into other systems
+2. Ensuring that entities inserted by Sesam are kept track of and merged with their origin entity 
+
+Point 1, e.g. a company represented in system *A* has a corresponding company represented in system *B* can be managed by by letting the entities merge in their global pipe. This also requires the identification of (preferably) uniquely identifying merge criterion that ensure these two entities are treated as the same entity. By applying the :ref:`namespace split pattern <namespace_split>` we can ensure that the correct version of the data is sent to the correct ``-share`` pipe.
```
Comment (djalil-bouhachi) [inline hub/bidirectional-synchronization.rst:75]: Typo: by by

File: hub/bidirectional-synchronization.rst (hunk 2 above)
Comment (erik-leven) [reply hub/bidirectional-synchronization.rst]: fixed

File: hub/bidirectional-synchronization.rst (hunk 3 above)
Comment (erik-leven) [reply hub/bidirectional-synchronization.rst]: fixed

File: hub/bidirectional-synchronization.rst [hunk 4]
```diff
@@ -52,17 +52,113 @@ Internal duplicate management
...
+3. Sesam updates system *A* without knowing about the changes done in step 2
+
+These situations can be managed by performing :ref:`optimistic locking <optimistic_locking>` in the ``-share`` pipe. This will ensure that the space in time where an entity *could* be updated without Sesam's knowledge is minimized. 
+
+Race condition management
+-------------------------
+
+Whenever you do bidirectional synchronization you have to be aware of potential  `race conditions <https://en.wikipedia.org/wiki/Race_condition>`__  when managing the data. 
+
+In Sesam we manage these race condition by ensuring the :ref:`completeness <completeness_feature>` of the data before letting it propagate downstream.
```
Comment (djalil-bouhachi) [inline hub/bidirectional-synchronization.rst:95]: Typo: these race condition"S"

File: hub/documentation/data-synchronization/establish-origin-pattern.rst [hunk 5]
```diff
@@ -0,0 +1,39 @@
+.. _establish_origin_pattern:
+
+Establish origin pattern
+------------------------
+
+In a :ref:`bidirectional synchronizaion <bidirectional-synchronization>` duplicate issues may arise if we no not know which entity from system *A* created which entity from system *B*. Without this relationship clearly established the new entity in system *B* might attempt to insert yet another entity into system *A* (because there might not be a connection between these two entities in the metadata).
```
Comment (djalil-bouhachi) [inline hub/documentation/data-synchronization/establish-origin-pattern.rst:6]: duplicate issues may arise if we do not know which entity in system B was created by system A.

File: hub/documentation/data-synchronization/establish-origin-pattern.rst [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+In Sesam we solve this issue with the *establish origin pattern*. This pattern has one external requirement:
+
+- When an entity is inserted into a system, the system's response has to include the new system specific primary key of the new entity. 
+
+To be able to establish the origin of an inserted entity inserts into system *A* need to be done through the :ref:` capture response with transform pattern <capture_response_with_transform>` which allows us to capture the response from the system, including the newly generated primary key
```
Comment (djalil-bouhachi) [inline hub/documentation/data-synchronization/establish-origin-pattern.rst]: Should it be system B here? Just to keep going on the same example

**Review by ellis-igneri — APPROVED** [review]

**Review by djalil-bouhachi — APPROVED** [review]

**Pull request merged**